clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/AllowLaunchAtomicOperation.groovy

```diff
@@ -153,6 +139,37 @@ class AllowLaunchAtomicOperation implements AtomicOperation<ResolvedAmiResult> {
     resolvedAmi
   }
 
+  private static enum ResolvedAmiLocation {
+    TARGET, SOURCE, THIRD_PARTY
+  }
+
+  private Tuple2<ResolvedAmiResult, ResolvedAmiLocation> resolveAmi(
```

Comment:
Tuple2 ok? seemed like a decent way to return 2 values with type information and destructure the result

Reply:
Yeah seems legit